text/0000-contingency-plan.md

```diff
@@ -0,0 +1,425 @@
+- Memory allocation in unsafe code is currently unstable. We expect to
+  be defining safe interfaces as part of the work on supporting
+  tracing garbage collectors (see [#415]).
+- The treatment of hygiene in macros is uneven (see [#22462], [#24278]). In some cases,
+  changes here may be backwards compatible, or may be more appropriate only with explicit opt-in
+  (or perhaps an alternate macro system altogether).
+- The layout of data structures is expected to change over time unless they are annotated
+  with a `#[repr(C)]` attribute.
+- Lints will evolve over time (both the lints that are enabled and the
+  precise cases that lints catch). We expect to introduce a
+  [means to limit the effect of these changes on dependencies][#1029].
+- Stack overflow is currently detected via a segmented stack check
+  prologue and results in an abort. We expect to experiment with a
+  system based on guard pages in the future.
+- We currently abort the process on OOM conditions (exceeding the heap space, overflowing
+  the stack). We may attempt to panic in such cases instead if possible.
+- Some details of type inference may change. For example, we expect to
+  implement the fallback mechanism described in [RFC 213], and we may
+  wish to make minor changes to accommodate overloaded integer
+  literals. In some cases, type inferences changes may be better
+  handled via explicit opt-in.
+
+(Although it is not directly covered by this RFC, it's worth noting in
+passing that some of the CLI flags to the compiler may change in the
+future as well. The `-Z` flags are of course explicitly unstable, but
+some of the `-C`, rustdoc, and linker-specific flags are expected to
+evolve over time.)
+
+### Opt-in changes
+
+For breaking changes that are not related to soundness or language
+semantics, but are still deemed desirable, an opt-in strategy can be
+used instead. This section describes an attribute for opting in to
+newer language updates, and gives guidelines on what kinds of changes
+should or should not be introduced in this fashion.
+
+We use the term *"opt-in changes"* to refer to changes that would be
+breaking changes, but are not because of the opt-in mechanism.
+
+#### Rust version attribute
+
+The specific proposal is an attribute `#![rust_version="X.Y"]` that
+can be attached to the crate; the version `X.Y` in this attribute is
+called the crate's "declared version". Every build of the Rust
+compiler will also have a version number built into it reflecting the
+current release.
+
+When a `#[rust_version="X.Y"]` attribute is encountered, the compiler
+will endeavor to produce the semantics of Rust "as it was" during
+version `X.Y`. RFCs that propose opt-in changes should discuss how the
+older behavior can be supported in the compiler, but this is expected
+to be straightforward: if supporting older behavior is hard to do, it
+may indicate that the opt-in change is too complex and should not be
+accepted.
```

Comment:
Sounds like a highway to code bloat. How long do we want to keep supporting a certain rust_version? Indefinitely?